cpp/src/plasma/plasma.cc

```diff
@@ -53,23 +53,15 @@ int warn_if_sigpipe(int status, int client_sock) {
   return -1;  // This is never reached.
 }
 
-/**
- * This will create a new ObjectInfo buffer. The first sizeof(int64_t) bytes
- * of this buffer are the length of the remaining message and the
- * remaining message is a serialized version of the object info.
- *
- * @param object_info The object info to be serialized
- * @return The object info buffer. It is the caller's responsibility to free
- *         this buffer with "delete" after it has been used.
- */
-std::unique_ptr<uint8_t[]> create_object_info_buffer(ObjectInfoT* object_info) {
+std::shared_ptr<std::vector<uint8_t>> create_object_info_buffer(
+      ObjectInfoT* object_info) {
   flatbuffers::FlatBufferBuilder fbb;
   auto message = CreateObjectInfo(fbb, object_info);
   fbb.Finish(message);
-  auto notification =
-      std::unique_ptr<uint8_t[]>(new uint8_t[sizeof(int64_t) + fbb.GetSize()]);
-  *(reinterpret_cast<int64_t*>(notification.get())) = fbb.GetSize();
-  memcpy(notification.get() + sizeof(int64_t), fbb.GetBufferPointer(), fbb.GetSize());
+  auto notification = std::make_shared<std::vector<uint8_t>>(
+      sizeof(int64_t) + fbb.GetSize());
+  *(reinterpret_cast<int64_t*>(notification->data())) = fbb.GetSize();
+  memcpy(notification->data() + sizeof(int64_t), fbb.GetBufferPointer(), fbb.GetSize());
```

Comment:
If you're creating a message buffer rather than a vector of independent values, wouldn't it make more sense to return a `std::string` (or perhaps a shared pointer thereof)?